src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -241,6 +289,9 @@ trait GraphDAO {
         // can't tell if this is a value or a map anymore so just remove both (there should only be one)
         vertex.removeProperty(key)
         vertex.getVertices(Direction.OUT, key).foreach(removeMapVertex)
+        //FIXME: ^^^ I think this is insufficient and doesn't cover the case where you overwrite a Some(foo:FooClass) with None, for a FooClass that requires sub-sub-vertices
+        //FIXME: See FIXME in getProperties - I think we can do this if we have the type too?
+        //Needs some shenanigans to figure out the type of an Option if it's None, see http://bit.ly/1htTZJk
```

Comment:
I think there's a similar issue to this going on in [line 299](https://github.com/broadinstitute/rawls/pull/90/files#diff-e295baee74fdedfd2e5dbfd08f2e7265R299); I don't think this isn't sufficient to clear up vertices left behind if we're overwriting a full list with an empty one.